src/ape_plugins/utils.py

```diff
@@ -298,6 +383,62 @@ def check_installed(self, use_cache: bool = True):
         ]
         return self.package_name in ape_packages
 
+    def _prepare_install(
+        self, upgrade: bool = False, skip_confirmation: bool = False
+    ) -> Optional[Dict[str, Any]]:
+        # NOTE: Internal and only meant to be called by the CLI.
+        if self.in_core:
+            logger.error(f"Cannot install core 'ape' plugin '{self.name}'.")
+            return None
+
+        elif self.version is not None and upgrade:
+            logger.error(
+                f"Cannot use '--upgrade' option when specifying "
+                f"a version for plugin '{self.name}'."
+            )
+            return None
+
+        # if plugin is installed but not trusted. It must be a third party
+        elif self.is_third_party:
+            logger.warning(f"Plugin '{self.name}' is not an trusted plugin.")
+
+        result_handler = ModifyPluginResultHandler(self)
+        pip_arguments = [sys.executable, "-m", "pip", "install"]
+
+        if upgrade:
+            logger.info(f"Upgrading '{self.name}' plugin ...")
+
+            # NOTE: A simple --upgrade flag may upgrade the plugin
+            # to a version outside Core Ape's. Thus, we handle it
+            # with a version-specifier instead.
+            # NOTE: There can issues when --quiet is not at the end.
```

Comment:
suggestion: Might be useful to note what issues could occur.

Reply:
It just can't be first, so comment was kinda confusing so i just removed it.